src/main/scala/com/wavesplatform/matcher/api/MatcherApiRoute.scala

```diff
@@ -73,13 +83,14 @@ case class MatcherApiRoute(wallet: Wallet,
   def orderBook: Route = (path("orderbook" / Segment / Segment) & get) { (a1, a2) =>
     parameters('depth.as[Int].?) { depth =>
       withAssetPair(a1, a2) { pair =>
-        onComplete((matcher ? GetOrderBookRequest(pair, depth)).mapTo[MatcherResponse]) {
-          case Success(resp) =>
-            resp.code match {
-              case StatusCodes.Found => redirect(Uri(s"/matcher/orderbook/$a2/$a1"), StatusCodes.Found)
-              case code              => complete(code -> resp.json)
+        getOrderBook(pair) match {
+          case Some(orderBook) =>
+            complete(StatusCodes.OK -> handleGetOrderBook(pair, orderBook, depth).json)
+          case None =>
+            getOrderBook(pair.reverse) match {
```

Comment:
Looks like this line won't be reached, because we are inside `withAssetPair`.
`withAssetPair` uses `createAssetPair`, that validates the pair and if it has a wrong order, we will fall to:
```scala
case Left(e)  => complete(StatusCodes.NotFound -> Json.obj("message" -> e))
```